### ISSUE TITLE
Connecting Curtain 3 in BLE

### ISSUE THREAD

**Issue by zdavid1999** [AdyRock/com.switchbot#150]
Hello,

I recently purchased the switchbot curtain 3 and I wanted to get it integrated with Homey Pro.
I did the initial setup using the switchbot app, and it works ok. I don't have a hub.

I can't setup the curtain 3 in Homey in BLE, so I choose HUB. I presume it is not working because I don't have one.
I therefore would like to switch to the BLE function, but I can't get it setup in Homey.

I try to add it, but it says it doesn't find any new device to link. I tried by inserting a pin in the Curtain 3 to get the led flashing, but I have the same result.

Am I missing something? How can I set it up using BLE?

Thanks,
David

**Comment by AdyRock** [comment]
It is probably out of range. Homey has a very short BLE range.

**Comment by AdyRock** [comment]
No, provided it's been set up in the SwitchBot app, Homey should detect it. 
But, as I say, Homey has a very poor BLE range, some worse than others. 
The app settings page has a log facility, so try setting the to 'Full information' and then try adding the device again. Then send the log, and I will take a look.

**Comment by zdavid1999** [comment]
Hello,

Thanks. I created a log 9029ec82-7822-4e78-bc53-4ee842492986

And also as you requested.

* 2024-12-01T10:30:41.146Z * BLE Discovery started * 2024-12-01T10:30:41.150Z * BLE HUB Discovery: [] * 2024-12-01T10:30:46.245Z * BLE Discovery found: [ { "_events": "...", "_eventsCount": "...", "_maxListeners": "...", "__peripheral": "...", "__client": "...", "__ble": "...", "id": "e3a419576231", "uuid": "e3a419576231", "address": "E3:A4:19:57:62:31", "addressType": "random", "connectable": true, "localName": "E3-A4-19-57-62-31", "manufacturerData": { "type": "Buffer", "data": [ 105, 9, 227, 164, 25, 87, 98, 49, 220, 11, 40, 17, 5, 0, 81 ] }, "serviceData": [ { "uuid": "0000fd3d-0000-1000-8000-00805f9b34fb", "data": { "type": "Buffer", "data": [ 91, 192, 81, 40, 17, 5 ] } } ], "serviceUuids": [], "rssi": -63, "timestamp": 1733049046243 }, { "_events": "...", "_eventsCount": "...", "_maxListeners": "...", "__peripheral": "...", "__client": "...", "__ble": "...", "id": "4b8b96818184", "uuid": "4b8b96818184", "address": "4B:8B:96:81:81:84", "addressType": "random", "connectable": true, "localName": "4B-8B-96-81-81-84", "manufacturerData": { "type": "Buffer", "data": [ 76, 0, 16, 6, 118, 30, 138, 97, 36, 41 ] }, "serviceData": [], "serviceUuids": [], "rssi": -46, "timestamp": 1733049046243 }, { "_events": "...", "_eventsCount": "...", "_maxListeners": "...", "__peripheral": "...", "__client": "...", "__ble": "...", "id": "6fadbf849224", "uuid": "6fadbf849224", "address": "6F:AD:BF:84:92:24", "addressType": "random", "connectable": true, "localName": "6F-AD-BF-84-92-24", "manufacturerData": { "type": "Buffer", "data": [ 76, 0, 16, 6, 10, 30, 74, 97, 143, 45 ] }, "serviceData": [], "serviceUuids": [], "rssi": -83, "timestamp": 1733049046243 }, { "_events": "...", "_eventsCount": "...", "_maxListeners": "...", "__peripheral": "...", "__client": "...", "__ble": "...", "id": "4eead2e230ba", "uuid": "4eead2e230ba", "address": "4E:EA:D2:E2:30:BA", "addressType": "random", "connectable": true, "localName": "4E-EA-D2-E2-30-BA", "manufacturerData": { "type": "Buffer", "data": [ 76, 0, 22, 8, 0, 44, 235, 52, 135, 96, 218, 51 ] }, "serviceData": [], "serviceUuids": [], "rssi": -54, "timestamp": 1733049046243 }, { "_events": "...", "_eventsCount": "...", "_maxListeners": "...", "__peripheral": "...", "__client": "...", "__ble": "...", "id": "f3f435709e8f", "uuid": "f3f435709e8f", "address": "F3:F4:35:70:9E:8F", "addressType": "random", "connectable": true, "localName": "F3-F4-35-70-9E-8F", "manufacturerData": { "type": "Buffer", "data": [ 76, 0, 18, 2, 0, 2 ] }, "serviceData": [], "serviceUuids": [], "rssi": -83, "timestamp": 1733049046244 } ] * 2024-12-01T10:30:46.247Z * BLE Device: E3:A4:19:57:62:31, "[", (-63) Serv: 5b c0 51 28 11 05 Manu: 69 09 e3 a4 19 57 62 31 dc 0b 28 11 05 00 51 * 2024-12-01T10:30:46.249Z * BLE Discovery finished


Thank you very much
David

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by zdavid1999** [comment]
Hello,

Did you have time to look at the log below?
I still can’t use my curtain opener with Homey :(

Thanks
David

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by BerndGaykema** [comment]
2 meters away is much too far. Mine only works within 50 cm.
But there is this ESP32 solution. As a hub, this ESP32 device makes the range much longer, in my case wel over 3 meters.
See [https://github.com/AdyRock/SwitchBotBLEHub](https://github.com/AdyRock/SwitchBotBLEHub) for the manual.

**Comment by AdyRock** [comment]
Unfortunately, all the formatting has been removed from the post, so it is really hard to read the jumble of data. Also I don't get the App Diagnostics Reports so I can't look there. From what I can make out, Homey has not picked up the service data, which is a sign of bad reception.
Just to be sure could you open the app settings page, select the Log page and set the Level to Full Information. Then try to add the device and go back to the settings page a tap on Send Log. I will get that log in a format that I can read.
